### PR TITLE
feat(view): add pulp::view::EditorBridge for renderer-agnostic editor↔processor JSON dispatch (#709, unblocks #468)

### PR DESCRIPTION
**Pull request by danielraffel**
Lifts Spectr's in-repo editor_bridge envelope/dispatcher into the
framework as `pulp::view::EditorBridge` so every plugin with an
editor — WebView today, native JS runtime via the #468 Claude Design
import lane tomorrow — shares one message-dispatch path instead of
reinventing envelope parsing, response building, and the standard
error vocabulary per plugin.

Framework owns:
  • envelope `{type, payload}` parse via choc::json
  • type → handler dispatch table with noexcept guarantees
  • response builders: ok_response() / ok_response(extras) / err_response(msg)
  • value-coercion helpers: get_float / get_uint / get_string (never throw)
  • standard envelope error vocabulary (substring-compatible with
    Spectr's existing test suite per #709 acceptance criterion):
      malformed_json, unknown_type, missing_field, wrong_type, internal_error
  • attach_webview(WebViewPanel&) wiring through set_message_handler
  • attach_native_runtime(JsRuntime&, name) stub for the #468 lane

Plugin keeps:
  • the set of message types and per-handler bodies
  • plugin-specific drag/edit state (carried in the handler closure)
  • domain-specific error strings via err_response("...")

CLI: `pulp import-design --from claude --file <path>` ingests a
manually-exported Claude Design HTML file (parses via the existing
Stitch HTML pipeline, tags IR as Claude) AND scaffolds a starter
`bridge_handlers.cpp` next to the generated JS view. Override path
with `--bridge-output`; skip with `--no-bridge-scaffold`. Per #468's
manual-export framing — no Anthropic API integration needed.

Skills + docs:
  • view-bridge skill: extended with EditorBridge section
  • import-design skill: claude source documented + bridge scaffold
    workflow
  • /pulp:import-design slash command: matching guidance
  • docs/reference/editor-bridge.md: full API + error vocabulary
  • docs/guides/webview.md: Host Bridge Dispatch section
  • skill_path_map.json: editor_bridge.{hpp,cpp} mapped to view-bridge;
    design_import.hpp added to import-design

Tests: test/test_editor_bridge.cpp — 24 cases, 55 assertions covering
all five envelope error categories, dispatch happy paths, response
builder shapes, value-coercion edge cases, and move semantics.

Spectr drop-in verified by porting Spectr's editor_bridge.cpp to
pulp::view::EditorBridge against this header — all 11 message types
map cleanly using only the framework's public API. Spectr-side cutover
PR (spectr#12) is the consumer-side follow-up.

Refs: #709, #468

### CODE REVIEW

**Comment by danielraffel** [comment]
PR looks great — reading the framework surface. Summary from the Spectr consumer side:

**✅ Surface matches the proposal verbatim.** Every method signature in the header is what I mirrored into the stand-in. Mechanical swap when this merges: delete \`host_bridge.{hpp,cpp}\` in Spectr, rename \`spectr::HostBridge → pulp::view::EditorBridge\`, swap one include, done.

**⚠ Minor: \`get_int\` not in this PR.** I asked on [#709](https://github.com/danielraffel/pulp/issues/709#issuecomment-4311448632); you may have deferred since Checkpoint 2 was already API-frozen. Not a blocker — Spectr's \`param_set\` handler uses \`get_uint\` since \`pulp::state::ParamID\` is \`uint32_t\` anyway. Fine to land this PR as-is and add \`get_int\` in a follow-up when the first real signed-int-valued payload field surfaces.

**✅ CLI scaffold + skills + slash command + docs in the same PR.** That's the full \"framework + tooling\" cutover-ready story from #709. No skill-sync gate surprises.

**✅ \`attach_native_runtime\` stub for #468.** Exactly right — #468's import lane can firm up the postMessage shape without blocking this.

I'll start drafting the Spectr cutover PR against this branch today so it's ready to land the moment this merges + the new SDK release ships. No material gaps to flag.

**Comment by danielraffel** [comment]
First shipyard run hit three failures, all environmental, none in the editor-bridge code:

1. **UBSan (macOS ARM64)** → test #989 `Clipboard data reflects pasteboard contents, not stale fallback` — known NSPasteboard contention flake. Same test passed on local rerun during preflight.
2. **ASan (macOS ARM64)** → test #210 `AudioFocusRegistry: current() is lock-free / audio-thread safe` — audio subsystem concurrency timing, unrelated to core/view changes.
3. **Windows shipyard SSH** → `Bundle apply failed: Remote bundle apply failed: #< CLIXML` — recurrence of the ssh-windows bundle-apply split-brain that #702 was supposed to close (Shipyard v0.40.0 bump). Namespace Windows lane PASSED (14m38s), and MSVC release-path gate PASSED (15m54s), so Windows *compiles* fine; it's the shipyard bundle infra that regressed.

Re-dispatched the two failing sanitizer jobs via `gh run rerun --failed` and retrying `shipyard ship` for the Windows SSH lane. Will check back when the cycle completes.

**Comment by danielraffel** [comment]
Pushed coverage fixup f6ef418f addressing the 41.8% diff-coverage failure:

- `test/test_design_import.cpp` — 3 assertions for the claude branch in parse_design_source / design_source_name + parse_claude_html delegation contract (was 14.3% on design_import diff).
- `test/test_editor_bridge.cpp` — stub WebViewPanel covers `attach_webview` routing; locally-defined JsRuntime covers the `attach_native_runtime` no-op stub for #468; new move-assignment test (was 86.3% on editor_bridge diff).
- `test/test_cli_shellout.cpp` — two end-to-end tests for `pulp import-design --from claude`: one asserts the bridge scaffold contains `pulp::view::EditorBridge` / `add_handler` / `attach_webview` literals, one asserts `--no-bridge-scaffold` opt-out (was 0% on the CLI tool diff).

All three updated test targets pass locally (65, 209, 2 assertions respectively). Re-running shipyard ship for the cross-platform lanes; GitHub Actions auto-reran on the push.

**Comment by danielraffel** [comment]
Pushed e7bc4724 to close the diff-coverage gap. Codecov on the prior SHA showed:

- design_import.cpp: 100% (claude branch + parser delegation)
- editor_bridge.cpp: 91.8% (remaining 12 lines are catch(...) defensive fallbacks)
- pulp_import_design.cpp: 0% (60-line scaffold body in CLI tool — codecov can't follow shellout subprocesses)

Aggregate was 60% / gate is 75%. Fix: moved `render_bridge_scaffold` from the CLI tool into `pulp::view::render_claude_bridge_scaffold` in `core/view/{include,src}/pulp/view/design_import.*`, unit-tested it directly (13 new assertions covering framework references, handler-registration patterns, attach call, and path interpolation), and had the CLI call into the library. Functionally identical — the CLI shellout test still passes against the same generated output.

Re-running shipyard ship with `--skip-target windows` since the SSH-Windows lane has a recurring `Bundle apply failed: Remote bundle apply failed: #< CLIXML` infrastructure error (independent of this PR; PR #702 was supposed to close it but didn't take fully). Namespace Windows lane and Windows MSVC release-path gate both pass on every run, so Windows IS validated end-to-end via Namespace — just the duplicate shipyard-local SSH path is broken.

File: core/view/src/editor_bridge.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        payload = choc::json::parse(payload_json);
+    } catch (...) {
+        return err_response("malformed JSON");
+    }
+    return dispatch(type, payload);
+}
+
+void EditorBridge::attach_webview(WebViewPanel& panel) {
+    panel.set_message_handler([this](const WebViewMessage& m) {
+        return dispatch_webview_message(m.type, m.payload_json);
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply core/view/src/editor_bridge.cpp:107]: Fixed in 1695ca6b. EditorBridge is now non-copyable AND non-movable (deleted all four special-member operations) so attach_webview's captured `this` can't dangle. static_asserts in test_editor_bridge.cpp lock the contract in. Removed the prior move-construct/move-assign tests since they no longer compile by design.

File: core/view/src/editor_bridge.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                                   const char* key,
+                                   std::size_t dflt) noexcept
+{
+    try {
+        if (!v.isObject() || !v.hasObjectMember(key)) return dflt;
+        const auto e = v[key];
+        if (e.isInt32())   { const auto x = e.getInt32();   return x < 0 ? std::size_t{0} : static_cast<std::size_t>(x); }
+        if (e.isInt64())   { const auto x = e.getInt64();   return x < 0 ? std::size_t{0} : static_cast<std::size_t>(x); }
+        if (e.isFloat64()) { const auto x = e.getFloat64(); return x < 0 ? std::size_t{0} : static_cast<std::size_t>(x); }
+        if (e.isFloat32()) { const auto x = e.getFloat32(); return x < 0 ? std::size_t{0} : static_cast<std::size_t>(x); }
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply core/view/src/editor_bridge.cpp]: Fixed in 1695ca6b. get_uint now clamps both ends before the cast: floats above SIZE_MAX (incl. 1e300 exponent territory) → std::numeric_limits<size_t>::max(); same for int64 on hypothetical 32-bit hosts. New `[issue-709]` test exercises 1e30 / 1e300 / huge f32 / -1e20.
